app/scripts/modules/core/src/pipeline/config/stages/common/ExecutionBarLabel.tsx

```diff
@@ -47,8 +48,8 @@ export class ExecutionBarLabel extends React.Component<IExecutionBarLabelProps,
 
   public render() {
     const { stage, application, execution, executionMarker } = this.props;
-    const inSuspendedExecutionWindow = stage.inSuspendedExecutionWindow;
-    if (inSuspendedExecutionWindow && executionMarker) {
+    const { suspendedStageTypes } = stage;
+    if (suspendedStageTypes.has('restrictExecutionDuringTimeWindow') && executionMarker) {
```

Comment:
I think if we add one more stage to the list of suspend-able types, it'll make sense to make the stages themselves register a handler for this marker treatment. Until then the `if`/`else` seems mostly ok to me.